### PR TITLE
mds/Server: replace ceph_assert for uid/gid with log prints

### PR DESCRIPTION
**Pull request by mihalicyn**
Having ceph_assert calls is unsafe in the Server::prepare_new_inode when assertion depends on a data which is a client-provided. A malicious client can easily crash MDS by purpose.

Let's replace two ceph_assert calls for owner_{u,g}id's with log prints.

See also:
https://github.com/ceph/ceph/pull/54108

Fixes: commit 46cb244b9c8397 ("ceph_fs.h: add separate owner_{u,g}id fields")
Fixes: https://tracker.ceph.com/issues/62217

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by mihalicyn** [comment]
cc @vshankar @lxbsz

**Comment by mihalicyn** [comment]
jenkins test make check

**Comment by mihalicyn** [comment]
jenkins test make check arm64

File: src/mds/Server.cc [hunk 1]
```diff
@@ -3517,11 +3517,15 @@ CInode* Server::prepare_new_inode(MDRequestRef& mdr, CDir *dir, inodeno_t useino
...
-    ceph_assert(_inode->gid != (unsigned)-1);
   }
 
   _inode->uid = mdr->client_request->get_owner_uid();
-  ceph_assert(_inode->uid != (unsigned)-1);
+
+  if (_inode->gid == (unsigned)-1 || _inode->uid == (unsigned)-1) {
+    dout(0) << "WARNING: client specified uid " << _inode->uid << " gid " << _inode->gid << " for ino " << _inode->ino << dendl;
+    mds->clog->error() << mdr->client_request->get_source()
+       << " specified uid " << _inode->uid << " gid " << _inode->gid << " for ino " << _inode->ino;
```
Comment (batrick) [inline src/mds/Server.cc:3527]: This could generate incredible amount of spam. Is this really an error or can we replace it with a reasonable default?

**Review by batrick — CHANGES_REQUESTED** [review]

File: src/mds/Server.cc (hunk 1 above)
Comment (mihalicyn) [reply src/mds/Server.cc:3527]: This is an error and if we have `uid`/`gid` equal to `-1` it means that something is terribly wrong with the client. Ideally, we need to fail the request somehow. Xiubo (@lxbsz) what do you think about this?

**Comment by vshankar** [comment]
@mihalicyn Do we need this change now since that actual crash is not fixed?

**Comment by mihalicyn** [comment]
Hi, Venky!

Hm, we have fixed the actual crash everywhere, so I guess we can drop this change. As you say.

**Pull request closed without merging**